### PR TITLE
Prototype work for `filter`.

### PR DESCRIPTION
**Pull request by jdunkerley**
### Checklist

Please include the following checklist in your PR:

- [ ] The documentation has been updated if necessary.
- [ ] All code conforms to the
      [Scala](https://github.com/enso-org/enso/blob/develop/docs/style-guide/scala.md),
      [Java](https://github.com/enso-org/enso/blob/develop/docs/style-guide/java.md),
      and
      [Rust](https://github.com/enso-org/enso/blob/develop/docs/style-guide/rust.md)
      style guides.
- All code has been tested:
  - [ ] Unit tests have been written where possible.
  - [ ] If GUI codebase was changed: Enso GUI was tested when built using BOTH
        `./run ide build` and `./run ide watch`.

### CODE REVIEW

File: distribution/lib/Standard/Table/0.0.0-dev/src/Data/Table.enso [hunk 1]
```diff
@@ -927,45 +947,6 @@ type Table
...
-
-       > Example
-         Remove any columns that contain missing values from the table.
-
-             import Standard.Examples
-
-             example_drop_missing_cols =
-                 Examples.inventory_table.drop_missing_columns
-    drop_missing_columns : Table
-    drop_missing_columns self =
```
Comment (radeusgd) [inline distribution/lib/Standard/Table/0.0.0-dev/src/Data/Table.enso:964]: This function is not accounted for by the new design. Are you sure we want to remove it?

Comment (jdunkerley) [reply distribution/lib/Standard/Table/0.0.0-dev/src/Data/Table.enso:964]: Felt a very odd function just to drop columns with NULL values and didn't feel right in it's current form.

It is fairly easy to do:
```python
table.select_columns By_Column (table.columns.filter c->(c.contains Nothing).not)
```

Comment (radeusgd) [reply distribution/lib/Standard/Table/0.0.0-dev/src/Data/Table.enso:964]: It was inspired by Pandas' [`dropna`](https://pandas.pydata.org/docs/reference/api/pandas.DataFrame.dropna.html) (with `axis=1` it does exactly that).

I remember using it sometimes - in some cases, we wanted to care only about columns that contain all values. Of course it's probably better to inspect the column and one may drop them manually - so I'm not saying this function is very needed. But it seems to be used from time to time, so may be worth considering. We can always re-add too it if a need arises.

Comment (jdunkerley) [reply distribution/lib/Standard/Table/0.0.0-dev/src/Data/Table.enso:964]: Ok will add it back in and amend design note.

**Pull request closed without merging**